src/test/java/pdfParsers/PdfToJsonConverterTest.java

```diff
@@ -0,0 +1,121 @@
+    public void textPdfHasExpectedFields() throws Exception {
+        assertThat(getFieldsFromPdf("text.pdf"),
+            contains(
+                allOf(
+                    hasJsonProperty("name", equalTo("single")),
+                    hasJsonProperty("value", equalTo("")),
+                    hasJsonProperty("type", equalTo("text")),
+                    hasJsonProperty("tabIndex", equalTo(0)),
+                    hasJsonProperty("required", equalTo(false)),
+                    hasJsonProperty("altText", equalTo(""))
+                ),
+                allOf(
+                    hasJsonProperty("name", equalTo("multiline")),
+                    hasJsonProperty("value", equalTo("")),
+                    hasJsonProperty("type", equalTo("text")),
+                    hasJsonProperty("tabIndex", equalTo(1)),
+                    hasJsonProperty("required", equalTo(false)),
+                    hasJsonProperty("altText", equalTo(""))
+                )
+            )
+        );
+    }
+
+    /**
+     * A matcher that verifies that the value of some specific property within the given JSON
+     * node has a value that passes the specified test.
+     */
+    private Matcher<JsonNode> hasJsonProperty(final String name, final Matcher<?> test) {
+        return new TypeSafeDiagnosingMatcher<JsonNode>() {
+            @Override
+            protected boolean matchesSafely(JsonNode item, Description mismatchDescription) {
+                JsonNode field = item.get(name);
+                Object value;
+                if (field.isBoolean()) {
+                    value = field.getValueAsBoolean();
+                } else if (field.isInt()) {
+                    value = field.getValueAsInt();
+                } else if (field.isDouble()) {
+                    value = field.getValueAsDouble();
+                } else {
+                    value = field.getValueAsText();
+                }
+                if (test.matches(value)) {
+                    return true;
+                } else {
+                    mismatchDescription.appendText("field \"" + name + "\": ");
+                    test.describeMismatch(value, mismatchDescription);
+                    return false;
+                }
+            }
+
+            @Override
+            public void describeTo(Description d) {
+                d.appendText("field \"" + name + "\": ");
+                test.describeTo(d);
+            }
+        };
+    }
+
+    /**
+     * Parses a PDF from the specified file (which must exist in src/test/resources);
+     * takes the resulting JSON data, parses that, and returns the array of fields from
+     * its "fields" property.
+     */
+    private ArrayNode getFieldsFromPdf(String fileName) throws IOException {
+        InputStream stream = getClass().getClassLoader().getResourceAsStream(fileName);
+        PdfToJsonConverter converter = new PdfToJsonConverter(new LocalPdfReader(stream));
+        String json = converter.getFields();
```

Comment:
This is a little awkward— the converter is building a JSON tree, then turning it into a string and returning the string, and then we're parsing the string back into a JSON tree again. It might be worthwhile to refactor things a bit so that this instead returns a data structure we can examine in detail without having to parse it (so for instance the list of fields would be an `Iterable<X>` where `X` is some class we create to contain the relevant field properties, translated from what IText gives us), and the unit tests can use that data structure... and then we have a separate component whose job is just to translate that data structure into JSON if necessary, and some other much simpler unit tests to verify that the JSONification is done correctly.